python/ray/experimental/state/api.py

```diff
@@ -1209,6 +1218,152 @@ def get_log(
             yield logs
 
 
+async def _download_file(
+    log_filename: str,
+    download_dir: str,
+    timeout: int,
+    api_server_url: str,
+    node_ip: Optional[str] = None,
+    node_id: Optional[str] = None,
+) -> DownloadLogFileResult:
+    """Helper function to download a single log file from a node
+
+    Args:
+        log_filename: File name relative to the log directory.
+        download_dir: Directory path to download the logs into.
+        timeout: Max timeout for requests made when downloading the logs.
+        api_server_url: API server URL where the download HTTP request is made to.
+        node_id: Id of the node containing the logs.
+        node_ip: Ip of the node containing the logs.
+
+    Return:
+        :class:`DownloadLogFileResult <ray.experimental.state.common.DownloadLogFileResult>`
+    """  # noqa: E501
+    options = dataclasses.asdict(
+        GetLogOptions(
+            node_id=node_id,
+            node_ip=node_ip,
+            filename=log_filename,
+            timeout=timeout,
+        )
+    )
+    # Remove the None values
+    options = {k: v for k, v in options.items() if v is not None}
+
+    # Open file
+    download_path = Path(f"{download_dir}/{log_filename}")
+
+    # Make sure parent dir exist for recursive files.
+    # Also, we are not cleaning this up if failure happens.
+    download_path.parent.mkdir(exist_ok=True, parents=True)
```

Comment:
Actually can we just create a folder and create files inside it? We can also delete the directory if it fails in that way?